benchmark/runner/mini_runners.cpp

```diff
@@ -681,42 +681,82 @@ static void GenUpdateDeleteIndexArguments(benchmark::internal::Benchmark *b) {
 /**
  * Arg <0, 1, 2, 3, 4, 5>
  * 0 - # integers to scan
- * 1 - # bigints to scan
+ * 1 - # mixed type to scan
  * 2 - # integers in table
- * 3 - # bigints in table
+ * 3 - # mixed type in table
  * 4 - row
  * 5 - cardinality
+ * 6 - mixed type is a varchar or not
+ * 7 - # possible threads
  */
 static void GenCreateIndexArguments(benchmark::internal::Benchmark *b) {
-  auto num_cols = {1, 3, 5, 7, 9, 11, 13, 15};
+  // 0 is a special argument used to indicate serial
+  auto num_threads = {0, 1, 2, 4, 8, 16};
+  auto num_cols = {1, 2, 3, 4, 5, 7, 9, 11, 13, 15};
   auto types = {type::TypeId::INTEGER, type::TypeId::BIGINT};
   std::vector<int64_t> row_nums = {1,    3,    5,     7,     10,    50,     100,    200,    500,    1000,
                                    2000, 5000, 10000, 20000, 50000, 100000, 200000, 500000, 1000000};
-  for (auto type : types) {
-    for (auto col : num_cols) {
-      for (auto row : row_nums) {
-        int64_t car = 1;
-        if (row > create_index_small_limit) {
-          // For these, we get a memory explosion if the cardinality is too low.
+  for (auto thread : num_threads) {
+    for (auto type : types) {
+      for (auto col : num_cols) {
+        for (auto row : row_nums) {
+          int64_t car = 1;
+          if (row > create_index_small_limit) {
+            // For these, we get a memory explosion if the cardinality is too low.
+            while (car < row) {
+              car *= 2;
+            }
+            car = car / (pow(2, create_index_large_cardinality_num));
+          }
+
           while (car < row) {
+            if (type == type::TypeId::INTEGER)
+              b->Args({col, 0, 15, 0, row, car, 0, thread});
+            else if (type == type::TypeId::BIGINT)
+              b->Args({0, col, 0, 15, row, car, 0, thread});
             car *= 2;
           }
-          car = car / (pow(2, create_index_large_cardinality_num));
-        }
 
-        while (car < row) {
           if (type == type::TypeId::INTEGER)
-            b->Args({col, 0, 15, 0, row, car});
+            b->Args({col, 0, 15, 0, row, row, 0, thread});
           else if (type == type::TypeId::BIGINT)
-            b->Args({0, col, 0, 15, row, car});
+            b->Args({0, col, 0, 15, row, row, 0, thread});
+        }
+      }
+    }
+  }
+}
+
+static void GenCreateIndexMixedArguments(benchmark::internal::Benchmark *b) {
+  // 0 is a special argument used to indicate serial
+  auto num_threads = {0, 1, 2, 4, 8, 16};
+  std::vector<int64_t> row_nums = {1,    3,    5,     7,     10,    50,     100,    200,    500,    1000,
+                                   2000, 5000, 10000, 20000, 50000, 100000, 200000, 500000, 1000000};
```

Comment:
These values repeat quite a lot. I think we would extract them into a common vector and also pass them into `TableGenerator` when we generate test tabels for `MiniRunners`. For now just use a global variable here (I'm thinking of creating another config file when I clean up the code). And then we can special case for the runners that don't have these exact parameters. Also, create a `GenerateMiniRunnerTables` function in `TableGenerator` to seperate the logic to create mini runner tables from to logic to create other test tables.